### DISCUSSION TITLE
Deleting wrongly detected laser points

### DISCUSSION THREAD

**Discussion by MonaFuhrmann** [biigle/core#618] [Q&A] [answered]
Hi,
I have submitted automated laser point detection, but find that there are a few wrong ones (detects laser points on images with none). I havent figured out how I simply delete these?
Mona

**Comment by mzur** [comment, marked as answer]
You can't delete the detected laser points but you can overwrite them. If you see that automatic detections are wrong, you can add manual detections and then rerun the laser point detection for that image (not for the whole volume). This can be done on the image information view (click the "i" icon of an image in the volume overview).

**Reply by MonaFuhrmann** [reply]
Thanks, yes, I have done that:) Unfortunately its very unpractical, being a lot of images and having to go to the overview to relocate the specific pictures. Hoping for some improvements, also with issues mentioned earlier of easy export of coordinates of lasers.

**Reply by mzur** [reply]
Do you have a suggestion how this could be improved? If you delete the wrong detections, you still need to reannotate and reprocess the laser points, right?

**Reply by MonaFuhrmann** [reply]
Somehow it would be easier if automatically detected laserpoints are treated as annotations (in the way that the manual ones are), then the deletion and export of coordinates x,y would be more straight forward - but maybe there is already an easy way to convert those into annotations (an I missed it?).
Another way could be to have the option of "resubmitting the laser detection" directly at the image level. Its quite difficult to redo/annotate lasers manually, then have to go out to the overview to find the picture, and then click the (i) in order to resubmit.

**Reply by mzur** [reply]
> Somehow it would be easier if automatically detected laserpoints are treated as annotations [...]

There is a [convert laser points](https://github.com/biigle/community-resources/tree/master/convert-laser-points) Python script in our community resources. This could be used to convert the automatic detections to annotations. The functionality of the script could be integrated in the user interface at some point (https://github.com/biigle/laserpoints/issues/59).

> Another way could be to have the option of "resubmitting the laser detection" directly at the image level.

This could be part of another idea that is tracked here: https://github.com/biigle/core/issues/620

Thanks for the feedback!

**Comment by BelenABYS** [comment]
Hi, I’m following up on this post as I encountered the same issue. I’m using the "Laser Point Detection" tool, but all the automatically detected laser points are incorrectly located. Based on @mzur's advice, I added manual detections and then reran the laser point detection. However, I am facing a challenge: many images without any lasers have been "annotated" by the "Automatic Laser Detection" tool. Since there are no actual lasers to annotate in these images, I don't know how to eliminate these false lasers.

I need to remove these automatic annotations from over 600 images. Is there an alternative solution to address this issue? I’m okay with deleting all the laser annotations, including the manually added lasers, if necessary.

I would greatly appreciate any guidance or suggestions. Thank you in advance for your help, @mzur!

**Reply by dlangenk** [reply]
Hi Belen,

I added an issue regarding the deletion of laserpoints (https://github.com/biigle/laserpoints/issues/88) and we're looking into finding a solution that will remove the laserpoints in the meantime.
I found one that will get rid of them, but the computed area still persists. However, I can set it to a fixed value if that helps. This can be done quite quickly. The solution would be to clone the whole volume (https://biigle.de/manual/tutorials/projects/about#volumes), which will remove the automatic, but not the manual laserpoints. Then update the metadata (https://biigle.de/manual/tutorials/volumes/file-metadata) with area set to a fixed size. Martin is busy today, because the Biigle beginners workshop was also today.

Best
Daniel

**Reply by BelenABYS** [reply]
Dear Daniel,
Thank you so much for your answer. I will try out the proposed temporal solution.

Best,
Belen

**Reply by mzur** [reply]
Please let me know how it worked out. If this is a one-time issue, we could also delete the LP information manually in the database.

**Comment by BelenABYS** [comment]
Dear Martin,

Your solution will be very helpful. The area of the images without lasers
is unknown, but the computed area related to the automatic lasers will
persist. In the solution proposed in a previous message the area will be
set to a fixed value instead of being deleted. If deleting the automatic
lasers and area is possible for you that will be excellent. It is for me a
onetime  issue as I will only annotate  lasers manually in the future.
Thank you in advance

All the best,

Belén

**Reply by mzur** [reply]
Ok please send me the link to your affected volume to info@biigle.de and we will sort this out there.

**Reply by dlangenk** [reply]
@BelenABYS Just keep in mind that even if you only manually annotate laserpoints and then run the area computation on the whole volume, it will automatically trigger the automatic laser point detection on the images without any laserpoints (if I am not mistaken @mzur)

**Reply by mzur** [reply]
This is correct (until https://github.com/biigle/laserpoints/issues/57 is implemented).

**Comment by BelenABYS** [comment]
Hi Martin,

This is the link https://biigle.de/volumes/19193 can you access it?

On Fri, Dec 6, 2024 at 7:44 AM Martin Zurowietz ***@***.***>
wrote:

> Ok please send me the link to your affected volume to ***@***.*** and
> we will sort this out there.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/biigle/discussions/618#discussioncomment-11483858>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BNIVCPN7OWHFFGXKPUBBUU32EGBJPAVCNFSM6AAAAABTA45EDCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBYGM4DKOA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Ana Belén Yánez-Suárez
National Geographic Explorer
<https://explorers.nationalgeographic.org/directory/ana-belen-yanez-suarez-1>
/ Ph.D. Candidate in deep-sea ecology
Marine Institute of Memorial University of Newfoundland
https://www.linkedin.com/in/ana-belen-yanez/

**Reply by mzur** [reply]
Considering Daniel's comment above, you should first annotate all laserpoints manually, then run the detection and only then we can delete the remaining automatic detections.

**Comment by BelenABYS** [comment]
Good point, Daniel. Excellent idea, Martin. Given this situation, I will
first manually annotate all the lasers that need to be annotated and follow
up with you once it's done.
Thank you so much,

Belen

On Fri, Dec 6, 2024 at 9:02 AM Martin Zurowietz ***@***.***>
wrote:

> Considering Daniel's comment above, you should first annotate all
> laserpoints manually, then run the detection and only then we can delete
> the remaining automatic detections.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/biigle/discussions/618#discussioncomment-11484582>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BNIVCPJ6PTHF3Z2CXPUMA5T2EGKOPAVCNFSM6AAAAABTA45EDCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBYGQ2TQMQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Ana Belén Yánez-Suárez
National Geographic Explorer
<https://explorers.nationalgeographic.org/directory/ana-belen-yanez-suarez-1>
/ Ph.D. Candidate in deep-sea ecology
Marine Institute of Memorial University of Newfoundland
https://www.linkedin.com/in/ana-belen-yanez/

**Comment by BelenABYS** [comment]
Hi!
I have finished annotating all the lasers for the shared folder.
Please let me know how deleting the lasers works whenever you can.

Thank you so much for your help.

Belen

On Fri, Dec 6, 2024 at 9:08 AM Ana Belen Yanez Suarez ***@***.***>
wrote:

> Good point, Daniel. Excellent idea, Martin. Given this situation, I will
> first manually annotate all the lasers that need to be annotated and follow
> up with you once it's done.
> Thank you so much,
>
> Belen
>
> On Fri, Dec 6, 2024 at 9:02 AM Martin Zurowietz ***@***.***>
> wrote:
>
>> Considering Daniel's comment above, you should first annotate all
>> laserpoints manually, then run the detection and only then we can delete
>> the remaining automatic detections.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/orgs/biigle/discussions/618#discussioncomment-11484582>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BNIVCPJ6PTHF3Z2CXPUMA5T2EGKOPAVCNFSM6AAAAABTA45EDCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBYGQ2TQMQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
> Ana Belén Yánez-Suárez
> National Geographic Explorer
> <https://explorers.nationalgeographic.org/directory/ana-belen-yanez-suarez-1>
> / Ph.D. Candidate in deep-sea ecology
> Marine Institute of Memorial University of Newfoundland
> https://www.linkedin.com/in/ana-belen-yanez/
>


-- 
Ana Belén Yánez-Suárez
National Geographic Explorer
<https://explorers.nationalgeographic.org/directory/ana-belen-yanez-suarez-1>
/ Ph.D. Candidate in deep-sea ecology
Marine Institute of Memorial University of Newfoundland
https://www.linkedin.com/in/ana-belen-yanez/

**Reply by mzur** [reply]
I have now performed a new detection process with 10 cm laser distance. After that I have deleted all laser information that was not based on manual annotations. Please check if everything is correct now.

**Comment by BelenABYS** [comment]
Hi Martin,

I downloaded the report, and the area seems to be calculated without a
problem.
I appreciate your help.

Belen

On Tue, Dec 10, 2024 at 7:33 AM Martin Zurowietz ***@***.***>
wrote:

> I have now performed a new detection process with 10 cm laser distance.
> After that I have deleted all laser information that was not based on
> manual annotations. Please check if everything is correct now.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/biigle/discussions/618#discussioncomment-11519339>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BNIVCPIKO3DIRWILZMYJPW32E3DBJAVCNFSM6AAAAABTA45EDCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJRHEZTGOI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Ana Belén Yánez-Suárez
National Geographic Explorer
<https://explorers.nationalgeographic.org/directory/ana-belen-yanez-suarez-1>
/ Ph.D. Candidate in deep-sea ecology
Marine Institute of Memorial University of Newfoundland
https://www.linkedin.com/in/ana-belen-yanez/
